pkg/odo/cli/init/init.go

```diff
@@ -129,32 +133,34 @@ func (o *InitOptions) Run() (err error) {
 		}
 	}()
 
-	destDevfile := filepath.Join(o.destDir, "devfile.yaml")
+	destDevfile := filepath.Join(o.contextDir, "devfile.yaml")
 	if o.InitParams.DevfilePath != "" {
 		err = o.downloadDirect(o.InitParams.DevfilePath, destDevfile)
 	} else {
-		err = o.downloadFromRegistry(o.InitParams.DevfileRegistry, o.InitParams.Devfile, o.destDir)
+		err = o.downloadFromRegistry(o.InitParams.DevfileRegistry, o.InitParams.Devfile, o.contextDir)
 	}
 	if err != nil {
 		return fmt.Errorf("Unable to download devfile: %w", err)
 	}
 
-	devfileObj, err := devfile.ParseAndValidateFromFile(destDevfile)
+	resolved := false
+	devfileObj, _, err := devfile.ParseDevfileAndValidate(parser.ParserArgs{Path: destDevfile, FlattenedDevfile: &resolved})
```

Comment:
Just so that we don't regress back to this issue, can add an integration test testing this part?